nft/nft_lib/src/main.sw

```diff
@@ -0,0 +1,139 @@
+library nft_lib;
+
+use std::{
+    address::Address,
+    assert::require,
+    constants::NATIVE_ASSET_ID,
+    chain::auth::{AuthError, Sender, msg_sender},
+    result::*,
+    revert::revert,
+};
+
+enum Error {
+    AddressCannotBeZero: (),
+    NFTNotInitalized: (),
+    OwnerAlreadySet: (),
+    OwnerCannotBeApproved: (),
+    SenderNotOwner: (), 
+}
+
+// TODO: Add mappings when supported for token ids
+
+struct Data {
+    owner: Address,
+    approved: Address,
+}
+
+storage {
+    data: Data,
+    state: u64,
+}
+
+/// Sets the inital owner for the NFT
+///
+/// # Panics
+///
+/// The function will panic when:
+/// - The NFT has already been initalized
+/// - The new owner is the 0 address
+pub fn mint_to_address(new_owner: Address) -> bool {
+    require(storage.state == 0, Error::OwnerAlreadySet);
+    require(new_owner.value != NATIVE_ASSET_ID, Error::AddressCannotBeZero);
+
+    storage.data = Data {
+        owner: new_owner, approved: ~Address::from(NATIVE_ASSET_ID)
+    };
+    storage.state = 1;
+    true
+}
+
+/// Burns the NFT
+///
+/// # Panics
+///
+/// The function will panic when:
+/// - The sender is not the owner
+/// - The NFT has not be initalized
+pub fn burn() -> bool {
+    require(storage.state == 1, Error::NFTNotInitalized);
+
+    let sender: Result<Sender, AuthError> = msg_sender();
+    if let Sender::Address(address) = sender.unwrap() { 
```

Comment:
I think you can turn this into a match.  You can checkout the mutlisig for an example.  Although I don't know if match if preferable here.

Reply:
Yes, matching is preferable